internal/cache/gomap/gomap.go

```diff
@@ -0,0 +1,202 @@
+package gomap
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"maps"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/zitadel/zitadel/internal/cache"
+)
+
+type mapCache[I, K comparable, V cache.Entry[I, K]] struct {
+	config   *cache.CacheConfig
+	indexMap map[I]*index[K, V]
+	logger   *slog.Logger
+}
+
+// NewCache returns an in-memory Cache implementation based on the builtin go map type.
+// Object values are stored as-is and there is no encoding or decoding involved.
+func NewCache[I, K comparable, V cache.Entry[I, K]](background context.Context, indices []I, config cache.CacheConfig) cache.PrunerCache[I, K, V] {
+	m := &mapCache[I, K, V]{
+		config:   &config,
+		indexMap: make(map[I]*index[K, V], len(indices)),
+		logger: slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+			AddSource: true,
+			Level:     slog.LevelError,
+		})),
+	}
+	if config.Log != nil {
+		m.logger = config.Log.Slog()
+	}
+	m.logger.InfoContext(background, "map cache logging enabled")
+
+	for _, name := range indices {
+		m.indexMap[name] = &index[K, V]{
+			config:  m.config,
+			entries: make(map[K]*entry[V]),
+		}
+	}
+	return m
+}
+
+func (c *mapCache[I, K, V]) Get(ctx context.Context, index I, key K) (value V, ok bool) {
+	i, ok := c.indexMap[index]
+	if ok {
+		entry, err := i.Get(key)
+		if err == nil {
+			c.logger.DebugContext(ctx, "map cache get", "index", index, "key", key)
+			return entry.value, true
+		}
+		if errors.Is(err, cache.ErrCacheMiss) {
+			c.logger.InfoContext(ctx, "map cache get", "err", err, "index", index, "key", key)
+			return value, false
+		}
+	}
+	c.logger.ErrorContext(ctx, "map cache get", "err", cache.NewIndexUnknownErr(index), "index", index, "key", key)
+	return value, false
```

Comment:
cant we early return here?
```suggestion
	i, ok := c.indexMap[index]
    if !ok {
        c.logger.ErrorContext(ctx, "map cache get", "err", cache.NewIndexUnknownErr(index), "index", index, "key", key)
	    return value, false
    }
	
    entry, err := i.Get(key)
    if err == nil {
        c.logger.DebugContext(ctx, "map cache get", "index", index, "key", key)
        return entry.value, true
    }
    if errors.Is(err, cache.ErrCacheMiss) {
        c.logger.InfoContext(ctx, "map cache get", "err", err, "index", index, "key", key)
        return value, false
    }

	c.logger.ErrorContext(ctx, "map cache get", "err", cache.NewIndexUnknownErr(index), "index", index, "key", key)
	return value, false
```